stdpopsim/annotations.py

```diff
@@ -0,0 +1,165 @@
+    :vartype url: str
+    :ivar id: species id
+    :vartype id: str
+    :ivar species: a `stdpopsim.species` instance
+    :ivar description: description of annotation file
+    :vartype description: str
+    """
+    url = attr.ib(default=None)
+    zarr_url = attr.ib(default=None)
+    species = attr.ib(default=None)
+    id = attr.ib(default=None)
+    file_name = attr.ib(default=None)
+    description = attr.ib(default=None)
+    citations = attr.ib(factory=list)
+    long_description = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        self.file_name = os.path.basename(self.zarr_url)
+
+    @property
+    def cache_dir(self):
+        return pathlib.Path(cache.get_cache_dir()) / "annotations"
+
+    @property
+    def species_cache_dir(self):
+        return self.cache_dir / self.species.id
+
+    def __str__(self):
+        s = "GTF Annotation:\n"
+        s += "\tspecies   = {}\n".format(self.species.name)
+        s += "\tid        = {}\n".format(self.id)
+        s += "\turl       = {}\n".format(self.url)
+        s += "\tzarr url       = {}\n".format(self.zarr_url)
+        s += "\tcached    = {}\n".format(self.is_cached())
+        s += "\tcache_dir = {}\n".format(self.species_cache_dir)
+        return s
+
+    def is_cached(self):
+        """
+        Returns True if this map is cached locally.
+        """
+        return os.path.exists(self.species_cache_dir)
+
+    def download(self):
+        """
+        Downloads the zarr from the source URL and stores it in the
+        cache directory. If the annotation directory already exists it is first
+        removed.
+        """
+        self.file_name = os.path.basename(self.zarr_url)
+        if self.is_cached():
+            logger.info(f"Clearing cache {self.species_cache_dir}")
+            for fn in os.listdir(self.species_cache_dir):
+                file_path = os.path.join(self.species_cache_dir, fn)
+                try:
+                    os.unlink(file_path)
+                except OSError:
+                    warnings.warn(f'could not unlink {file_path}')
+        logger.debug(f"Checking species cache directory {self.species_cache_dir}")
+        os.makedirs(self.species_cache_dir, exist_ok=True)
+        download_file = f'{self.species_cache_dir}/{self.file_name}'
+        print(download_file)
+        logger.info(f"Downloading Zarr file '{self.id}' from {self.zarr_url}")
+        if os.path.exists(download_file):
+            warnings.warn("multiple downloads?")
+        try:
+            x, y = urllib.request.urlretrieve(self.zarr_url, download_file)
+        except FileNotFoundError:
+            warnings.warn("can't connnect to url")
+
+        logger.debug("Download Zarr complete")
+        logger.info(f"Storing Zarr in {self.species_cache_dir}")
+
+    def get_chromosome_annotations(self, id):
+        """
+        Returns the pandas dataframe for
+        the chromosome with the specified id.
```

Comment:
I suppose this will be the main entry point for consumers of annotations. So we'll need to have complete docstrings for this (and the other user-facing methods in this class). Happy for this to be punted down the road though. Maybe open a ticket tracking this once it's merged?